pkg/prometheus/statefulset.go

```diff
@@ -378,16 +379,25 @@ func makeStatefulSetSpec(p monitoringv1.Prometheus, c *Config, ruleConfigMapName
 				promArgs = append(promArgs,
 					fmt.Sprintf("-storage.tsdb.retention.size=%s", p.Spec.RetentionSize),
 				)
+			} else if p.Spec.Retention == "" {
+				p.Spec.Retention = defaultRetention
```

Comment:
I would prefer not to modify this field here, but rather create a new variable for this purpose.